### PR TITLE
[Bug] Adjust rename function

### PR DESCRIPTION
**Pull request by Corepex**
We should use the same default value for all rename operations ...

### CODE REVIEW

File: src/Controller/Admin/DataObject/DataObjectController.php [hunk 1]
```diff
@@ -1162,8 +1162,11 @@ private function executeUpdateAction(DataObject $object, mixed $values): array
                 return ['success' => false, 'message' => "You can't rename or relocate if there's a newer not published version"];
             }
         }
-        
-        $key = Service::getValidKey($values['key'], 'object') ?? null;
+
+        $key = $values['key'] ?? null;
+        if($key) {
+            $key = Service::getValidKey($values['key'], 'object');
```
Comment (blankse) [inline src/Controller/Admin/DataObject/DataObjectController.php]:
```suggestion
        if ($key) {
            $key = Service::getValidKey($key, 'object');
```

Comment (Corepex) [reply src/Controller/Admin/DataObject/DataObjectController.php]: thx

File: src/Controller/Admin/DataObject/DataObjectController.php [hunk 2]
```diff
@@ -1163,6 +1164,9 @@ private function executeUpdateAction(DataObject $object, mixed $values): array
         }
 
         $key = $values['key'] ?? null;
+        if ($key) {
+            $key = Service::getValidKey($key, 'object');
+        }
```
Comment (dvesh3) [inline src/Controller/Admin/DataObject/DataObjectController.php:1169]:
```suggestion
```
just found out that we handle it on client side https://github.com/pimcore/admin-ui-classic-bundle/blob/1976b8f869ad450e1b3d2b9e4aebcc47f3487e7e/public/js/pimcore/elementservice.js#L459

File: public/js/pimcore/asset/tree.js [hunk 3]
```diff
@@ -1184,7 +1184,7 @@
              elementType: "asset",
              elementSubType: record.data.type,
              id: record.data.id,
-             default: Ext.util.Format.htmlDecode(record.data.text)
+             default: record.data.key
```
Comment (dvesh3) [inline public/js/pimcore/asset/tree.js]: please revert as this is required https://github.com/pimcore/pimcore/pull/5206/files#diff-9d9bcbb5d9a089108c6e32c3b3885884382b70ce63550e82ddebbc81d33a39eeR310

File: public/js/pimcore/object/tree.js [hunk 4]
```diff
@@ -1119,7 +1119,7 @@ pimcore.registerNS("pimcore.object.tree");
              elementType: "object",
              elementSubType: record.data.type,
              id: record.data.id,
-             default: Ext.util.Format.htmlDecode(record.data.text)
+             default: record.data.key
```
Comment (dvesh3) [inline public/js/pimcore/object/tree.js]: please revert as this is required https://github.com/pimcore/pimcore/pull/5206/files#diff-9d9bcbb5d9a089108c6e32c3b3885884382b70ce63550e82ddebbc81d33a39eeR310

File: src/Controller/Admin/DataObject/DataObjectController.php (hunk 2 above)
Comment (Corepex) [reply src/Controller/Admin/DataObject/DataObjectController.php:1169]: is client-side enough? i guess we should also handle it on the server or?

File: public/js/pimcore/asset/tree.js (hunk 3 above)
Comment (Corepex) [reply public/js/pimcore/asset/tree.js]: done with 7131182ed24c7c1a40acce25e2fabc411a298da7

File: public/js/pimcore/object/tree.js (hunk 4 above)
Comment (Corepex) [reply public/js/pimcore/object/tree.js]: done with 7131182ed24c7c1a40acce25e2fabc411a298da7

File: src/Controller/Admin/DataObject/DataObjectController.php (hunk 2 above)
Comment (dvesh3) [reply src/Controller/Admin/DataObject/DataObjectController.php:1169]: yes, right.

**Pull request merged**